### DISCUSSION TITLE
New TestClient crashes

### DISCUSSION THREAD

**Discussion by Goldziher** [encode/starlette#1869] [Potential Issue] [answered]
Here is the stack trace:

```shell
tests/test_dependency.py:None (tests/test_dependency.py)
test_dependency.py:9: in <module>
    from starlite.testing import create_test_client
../starlite/testing.py:6: in <module>
    from starlette.testclient import TestClient as StarletteTestClient
../../../Library/Caches/pypoetry/virtualenvs/starlite-s4i-xLL2-py3.10/lib/python3.10/site-packages/starlette/testclient.py:353: in <module>
    class TestClient(httpx.Client):
../../../Library/Caches/pypoetry/virtualenvs/starlite-s4i-xLL2-py3.10/lib/python3.10/site-packages/starlette/testclient.py:405: in TestClient
    ) -> typing.Union[bool, httpx._client.UseClientDefault]:
E   AttributeError: module 'httpx._client' has no attribute 'UseClientDefault'
```

FYI @Kludex

**Comment by Kludex** [comment]
What version?

**Reply by Goldziher** [reply]
0.21.0

**Reply by Kludex** [reply]
httpx version...

**Reply by Goldziher** [reply]
Latest

**Reply by adriangb** [reply]
That attribute is there in the latest version of httpx (it has been since [June 2021](https://github.com/encode/httpx/releases/tag/0.18.2)). It looks like this is happening at import time which means it doesn't depend on the codepath being run and we certainly import (and use) TestClient in our tests without encountering the error you are reporting.

**Reply by Kludex** [reply]
So, is this an issue?

**Reply by notypecheck** [reply]
Works with  `starlette = "^0.21.0"`, `httpx = "^0.23.0"`

**Reply by adriangb** [reply]
https://colab.research.google.com/drive/1faUaUGyJQ4fAqN_jegLikdbfulS3L9fI

Works with `pip install starlette==0.21.0 httpx==0.22.0`

**Comment by Goldziher** [comment, marked as answer]
yup, my bad. cheers.
